recipes/cpython/all/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+cmake_minimum_required(VERSION 2.8.11)
+project(test_package C)
+
+include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+conan_basic_setup()
+
+find_package(PythonInterp REQUIRED)
+find_package(PythonLibs REQUIRED)
+
+string(FIND "${PYTHON_EXECUTABLE}" "${CONAN_CPYTHON_ROOT}" ROOT_SUBPOS)
+if(ROOT_SUBPOS EQUAL -1)
+    message(FATAL_ERROR "found wrong python interpreter: ${PYTHON_EXECUTABLE}")
+endif()
+
+message(STATUS "PYTHON_VERSION_STRING: ${PYTHON_VERSION_STRING}")
+
+if(NOT PYTHON_VERSION_STRING MATCHES "${Python_ADDITIONAL_VERSIONS}.*")
+    message("Python_ADDITIONAL_VERSIONS does not match PYTHON_VERSION_STRING")
+    message(FATAL_ERROR "CMake detected wrong cpython version")
+endif()
+
+add_library(spam MODULE test_module.c)
```

Comment:
Can we use more modern CMake, `find_package (Python3)` and [`Python3_add_library`](https://cmake.org/cmake/help/latest/module/FindPython3.html#commands) instead? It has [already implemented](https://github.com/Kitware/CMake/blob/master/Modules/FindPython/Support.cmake#L2767) the things like `.pyd`.

Reply:
Probably (yes), if you have a recent enough cmake version.
I used the older `FindPythonIterp.cmake` and `FindPythonLibs.cmake` for these older cmake versions.
But it is indeed useful to also test the newer `FindPython.cmake` module.

Reply:
This is done now.